types/applications/sidebarTabs/combatTracker.d.ts

```diff
@@ -0,0 +1,183 @@
+
+  /**
+   * Initialize the combat tracker to display a specific combat encounter.
+   * If no encounter is provided, the tracker will be initialized with the first encounter in the viewed scene.
+   * @param combat - The combat encounter to initialize
+   *                 (default: `null`)
+   * @param render - Whether to re-render the sidebar after initialization
+   *                 (default: `true`)
+   */
+  initialize({ combat, render }?: { combat?: Combat; render?: boolean }): void;
+
+  /**
+   * Scroll the combat log container to ensure the current Combatant turn is centered vertically
+   */
+  scrollToTurn(): void;
+
+  /**
+   * @override
+   */
+  getData(options?: Application.RenderOptions): CombatTracker.Data;
+
+  /**
+   * @override
+   */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Handle new Combat creation request
+   */
+  protected _onCombatCreate(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle a Combat deletion request
+   * @remarks This is never called
+   */
+  protected _onCombatDelete(event: Event): Promise<void>;
+
+  /**
+   * Handle a Combat cycle request
+   */
+  protected _onCombatCycle(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle click events on Combat control buttons
+   * @param event - The originating mousedown event
+   */
+  protected _onCombatControl(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle a Combatant control toggle
+   * @param event - The originating mousedown event
+   */
+  protected _onCombatantControl(event: JQuery.ClickEvent): Promise<void>;
+
+  /**
+   * Handle toggling the defeated status effect on a combatant Token
+   * @param c - The combatant data being modified
+   */
+  protected _onToggleDefeatedStatus(c: Combat.Combatant): Promise<void>;
+
+  /**
+   * Handle mouse-down event on a combatant name in the tracker
+   * @param event - The originating mousedown event
+   * @returns A Promise that resolves once the pan is complete
+   */
+  protected _onCombatantMouseDown(event: JQuery.ClickEvent): Promise<void> | void;
+
+  /**
+   * Handle mouse-hover events on a combatant in the tracker
+   */
+  protected _onCombatantHover(event: JQuery.MouseEnterEvent): void;
+
+  /**
+   * Handle mouse-unhover events for a combatant in the tracker
+   */
+  protected _onCombatantHoverOut(event: JQuery.MouseLeaveEvent): void;
+
+  /**
+   * Default folder context actions
+   */
+  protected _contextMenu(html: JQuery): void;
+
+  /**
+   * Get the sidebar directory entry context options
+   * @returns The sidebar entry context options
+   */
+  protected _getEntryContextOptions(): ContextMenu.Item[];
+
+  /**
+   * Display a dialog which prompts the user to enter a new initiative value for a Combatant
+   */
+  protected _onConfigureCombatant(li: JQuery): void;
+}
+
+declare namespace CombatTracker {
+  type Data = {
+    user: User;
+    started: undefined;
+    settings: any; // TODO type when Settings is typed
```

Comment:
I think this should be of type
```ts
{
  resource?: string;
  skipDefeated?: boolean;
}
```

Reply:
Yes. It may also be prudent to add [key: string]: unknown, since I believe you can add additional keys

Reply:
I will add this in my next PR. I think I will tackle the settings in general.